File: Python/offer32-2.py
# BFS
class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
      if not root: return []
      res, queue = [], collections.deque()
      queue.append(root)
      while queue:
        tmp = []
        for _ in range(len(queue)):
          node = queue.popleft()
          tmp.append(node.val)
          if node.left: queue.append(node.left)
          if node.right: queue.append(node.right)
        res.append(tmp)
      return res

File: Javascript/offer32-2.js
// BFS
var levelOrder = function(root) {
  if (!root) return [];
  const queue = [root];
  const res = [];
  let level = 0;
  while (queue.length) {
    res[level] = [];
    let levelNum = queue.length;
    while (levelNum--) {
      const node = queue.shift();
      res[level].push(node.val);
      if (node.left) queue.push(node.left)
      if (node.right) queue.push(node.right)
    }
    level++;
  }
  return res;
};